gui/options-dialog.cpp

```diff
@@ -163,3 +169,23 @@ void OptionsDialog::doCancel()
     close();
 }
 
+void OptionsDialog::browse_datalogging_file()
+{
+    QString filename = ui.tracklogging_filenameedit->text();
+    if (filename.isEmpty())
+        filename = QDir::currentPath();
+    /* Sometimes this function freezes the app before opening the dialog.
```

Comment:
@sthalik well there is this little issue though. This should really be idiot proof but it froze twice already.
I will continue using it with the non-native dialog with which it might work.